### PR TITLE
Refactor cheats window widget definitions

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
This refactors the cheats window to use groupbox-based offsets for its widget positions, rather than its custom macro system. A major benefit is widget flexibility without resorting to hacky positioning. This was previously seen in the use of decimal grid positions and added pixel offsets.

On top of this, by making the window adhere to the conventions used in the rest of the code base, this PR hopefully increases legibility without compromising widget mobility.

I hope to do some more redesigning of the cheats window in another PR after this.

### CODE REVIEW

File: src/openrct2-ui/windows/Cheats.cpp [hunk 1]
```diff
@@ -1184,30 +1178,30 @@ static void window_cheats_paint(rct_window *w, rct_drawpixelinfo *dpi)
...
-        gfx_draw_string_right(dpi, STR_FORMAT_MONTH,                &actual_month, w->colours[1], w->x + WPL(1) - 10 - TXTO, w->y + YPL(8) + TXTO);
-        gfx_draw_string_right(dpi, STR_FORMAT_INTEGER,              &day_spinner_value, w->colours[1], w->x + WPL(1) - 10 - TXTO, w->y + YPL(9) + TXTO);
+        gfx_draw_string_left(dpi, STR_BOTTOM_TOOLBAR_CASH, gCommonFormatArgs, colour, w->x + 11 + 3, w->y + 90 + 3);
+        gfx_draw_string_left(dpi, STR_YEAR, nullptr, COLOUR_BLACK, w->x + 11 + 3, w->y + 195 + 3);
+        gfx_draw_string_left(dpi, STR_MONTH, nullptr, COLOUR_BLACK, w->x + 11 + 3, w->y + 216 + 3);
+        gfx_draw_string_left(dpi, STR_DAY, nullptr, COLOUR_BLACK, w->x + 11 + 3, w->y + 237 + 3);
+        gfx_draw_string_right(dpi, STR_FORMAT_INTEGER, &year_spinner_value, w->colours[1], w->x + 226 - 3, w->y + 195 + 3);
+        gfx_draw_string_right(dpi, STR_FORMAT_MONTH, &actual_month, w->colours[1], w->x + 226 - 3, w->y + 216 + 3);
+        gfx_draw_string_right(dpi, STR_FORMAT_INTEGER, &day_spinner_value, w->colours[1], w->x + 226 - 3, w->y + 237 + 3);
```
Comment (Broxzier) [inline src/openrct2-ui/windows/Cheats.cpp]: I see many `11 + 3` here, why are they separate? Marging and padding?

**Comment by AaronVanGeffen** [comment]
> The macros had as advantage that it's easier to place widgets, making it more flexible than the hardcoded values. The XPL and YPL macros would give the position for a given row and column for example. This could be extended instead, to take group boxes and widget-specific paddings/margins into account.

I appreciate the argument, but disagree that such a grid system makes placement more flexible. Instead, I think it's constraining the window design. As a result, the layout of the cheat window looks very differently spaced from the rest of the game...

File: src/openrct2-ui/windows/Cheats.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/Cheats.cpp]: Indeed — the result of manually expanding macros. Looks like I've forgotten to combine these values.

**Comment by AaronVanGeffen** [comment]
I really don't like the current macro system, but I'll leave it to someone else to improve it, then. Thanks.

**Pull request closed without merging**